packages/deployments/contracts/contracts/core/connext/libraries/SwapUtils.sol

```diff
@@ -452,15 +449,15 @@ library SwapUtils {
         ++i;
       }
     }
-    c = c.mul(d).mul(AmplificationUtils.A_PRECISION).div(nA.mul(numTokens));
-    uint256 b = s.add(d.mul(AmplificationUtils.A_PRECISION).div(nA));
+    c = (c * d * (AmplificationUtils.A_PRECISION)) / (nA * numTokens);
+    uint256 b = s + ((d * AmplificationUtils.A_PRECISION) / nA);
     uint256 yPrev;
     uint256 y = d;
 
     // iterative approximation
     for (uint256 i; i < MAX_LOOP_LIMIT; ) {
       yPrev = y;
-      y = y.mul(y).add(c).div(y.mul(2).add(b).sub(d));
+      y = (y * y + c) / (y * 2 + b - d);
```

Comment:
I think `y = ((y * y) + c) / ((y * 2) + b - d);` would be safer

Reply:
hmm. looks good